packages/uploadthing/src/internal/client.ts

```diff
@@ -63,6 +63,11 @@ export const progressWidths: Record<number, string> = {
   100: "after:w-[100%]",
 };
 
+type MinCallbackArg = { ready: boolean | (() => boolean) };
+type inferRuntime<T extends MinCallbackArg> = T["ready"] extends boolean
+  ? "react"
+  : "solid";
```

Comment:
Oh, that's smart

Reply:
Not sure how well it scales though if we add Vue and Svelte support 😅 

There's probably a more universal way but this was the first idea i got

Reply:
Yeah, you are right, but I guess we can deal with it later. Not sure about Svelte PR, but the Vue one does not have custom styling feature

Though looking at your implementation, I have an idea. Is there a way to define a property on type, that will be hidden from user?

I mean can we have something like following?

```tsx
type SolidButtonProps = {
  /**
  * @private
  */
  __runtime: "solid";
  ...
}

type ReactButtonProps = {
  /**
  * @private
  */
  __runtime: "react";
  ...
}

type Runtime<T extends { __runtime: "react" | "solid" }> = T["__runtime"] extends "react" ? "react" : "solid"
```

With such we could specify type of runtime right on type of the callback args. Hope the idea is clear